cmd/server-main.go

```diff
@@ -226,9 +233,59 @@ func serverCmdArgs(ctx *cli.Context) []string {
 	return strings.Fields(v)
 }
 
-func serverHandleCmdArgs(ctx *cli.Context) {
-	// Handle common command args.
-	handleCommonCmdArgs(ctx)
+type configFileOpts struct {
+	FTP  map[string]string `yaml:"ftp"`
+	SFTP map[string]string `yaml:"sftp"`
+}
+
+type configFileMain struct {
+	Version     string         `yaml:"version"`
+	Addr        *string        `yaml:"address"`
+	ConsoleAddr *string        `yaml:"console-address"`
+	CertsDir    *string        `yaml:"certs-dir"`
+	Pools       [][]string     `yaml:"pools"`
+	Options     configFileOpts `yaml:"options"`
+}
```

Comment:
Also move this struct to `internal/config/server.go`

```go

// ServerConfig represents server command line options, server arguments
// and server config all in a single data structure.
type ServerConfig struct {
	Version        string     `yaml:"version"`
	Address        string     `yaml:"address"`
	ConsoleAddress string     `yaml:"console-address"`
	CertsDir       string     `yaml:"certs-dir"`
	Pools          [][]string `yaml:"pools"`
	Options        struct {
		Site struct {
			Name    string `yaml:"name"`
			Region  string `yaml:"region"`
			Comment string `yaml:"comment"`
		} `yaml:"site"`
		Ftp struct {
			Address          string `yaml:"address"`
			PassivePortRange string `yaml:"passive-port-range"`
		} `yaml:"ftp"`
		Sftp struct {
			Address       string `yaml:"address"`
			SSHPrivateKey string `yaml:"ssh-private-key"`
		} `yaml:"sftp"`
		Subnet struct {
			License string `yaml:"license"`
			APIKey  string `yaml:"api_key"`
			Proxy   string `yaml:"proxy"`
		} `yaml:"subnet"`
		Callhome struct {
			Frequency string `yaml:"frequency"`
		} `yaml:"callhome"`
		API struct {
			RequestsMax                 int    `yaml:"requests_max"`
			RequestsDeadline            string `yaml:"requests_deadline"`
			ClusterDeadline             string `yaml:"cluster_deadline"`
			CorsAllowOrigin             string `yaml:"cors_allow_origin"`
			RemoteTransportDeadline     string `yaml:"remote_transport_deadline"`
			ListQuorum                  string `yaml:"list_quorum"`
			ReplicationPriority         string `yaml:"replication_priority"`
			TransitionWorkers           string `yaml:"transition_workers"`
			StaleUploadsCleanupInterval string `yaml:"stale_uploads_cleanup_interval"`
			StaleUploadsExpiry          string `yaml:"stale_uploads_expiry"`
			DeleteCleanupInterval       string `yaml:"delete_cleanup_interval"`
			Odirect                     string `yaml:"odirect"`
			GzipObjects                 string `yaml:"gzip_objects"`
			RootAccess                  string `yaml:"root_access"`
			SyncEvents                  string `yaml:"sync_events"`
		} `yaml:"api"`
		Scanner struct {
			Speed string `yaml:"speed"`
		} `yaml:"scanner"`
		Compression struct {
			AllowEncryption string `yaml:"allow_encryption"`
			Extensions      string `yaml:"extensions"`
			MimeTypes       string `yaml:"mime_types"`
		} `yaml:"compression"`
		StorageClass struct {
			Standard string `yaml:"standard"`
			Rrs      string `yaml:"rrs"`
		} `yaml:"storage_class"`
		Heal struct {
			Bitrotscan   string `yaml:"bitrotscan"`
			MaxSleep     string `yaml:"max_sleep"`
			MaxIo        string `yaml:"max_io"`
			DriveWorkers string `yaml:"drive_workers"`
		} `yaml:"heal"`
		Etcd struct {
			Endpoints     string `yaml:"endpoints"`
			PathPrefix    string `yaml:"path_prefix"`
			CorednsPath   string `yaml:"coredns_path"`
			ClientCert    string `yaml:"client_cert"`
			ClientCertKey string `yaml:"client_cert_key"`
		} `yaml:"etcd"`
		IdentityTLS struct {
			SkipVerify string `yaml:"skip_verify"`
		} `yaml:"identity_tls"`
		IdentityPlugin struct {
			URL        string `yaml:"url"`
			AuthToken  string `yaml:"auth_token"`
			RolePolicy string `yaml:"role_policy"`
			RoleID     string `yaml:"role_id"`
		} `yaml:"identity_plugin"`
		PolicyPlugin struct {
			URL         string `yaml:"url"`
			AuthToken   string `yaml:"auth_token"`
			EnableHTTP2 string `yaml:"enable_http2"`
		} `yaml:"policy_plugin"`
		IdentityOpenid []struct {
			Name               string `yaml:"name"`
			Enable             string `yaml:"enable"`
			DisplayName        string `yaml:"display_name"`
			ConfigURL          string `yaml:"config_url"`
			ClientID           string `yaml:"client_id"`
			ClientSecret       string `yaml:"client_secret"`
			ClaimName          string `yaml:"claim_name"`
			ClaimUserinfo      string `yaml:"claim_userinfo"`
			RolePolicy         string `yaml:"role_policy"`
			ClaimPrefix        string `yaml:"claim_prefix"`
			RedirectURI        string `yaml:"redirect_uri"`
			RedirectURIDynamic string `yaml:"redirect_uri_dynamic"`
			Scopes             string `yaml:"scopes"`
			Vendor             string `yaml:"vendor"`
			KeycloakRealm      string `yaml:"keycloak_realm"`
			KeycloakAdminURL   string `yaml:"keycloak_admin_url"`
		} `yaml:"identity_openid"`
		IdentityLdap []struct {
			Name               string `yaml:"name"`
			Enable             string `yaml:"enable"`
			ServerAddr         string `yaml:"server_addr"`
			SrvRecordName      string `yaml:"srv_record_name"`
			UserDnSearchBaseDn string `yaml:"user_dn_search_base_dn"`
			UserDnSearchFilter string `yaml:"user_dn_search_filter"`
			GroupSearchFilter  string `yaml:"group_search_filter"`
			GroupSearchBaseDn  string `yaml:"group_search_base_dn"`
			TLSSkipVerify      string `yaml:"tls_skip_verify"`
			ServerInsecure     string `yaml:"server_insecure"`
			ServerStarttls     string `yaml:"server_starttls"`
			LookupBindDn       string `yaml:"lookup_bind_dn"`
			LookupBindPassword string `yaml:"lookup_bind_password"`
		} `yaml:"identity_ldap"`
		LoggerTerminal struct {
			Quiet     bool `yaml:"quiet"`
			Anonymous bool `yaml:"anonymous"`
			JSON      bool `yaml:"json"`
		} `yaml:"logger_terminal"`
		LoggerWebhook []struct {
			Name       string `yaml:"name"`
			Enable     string `yaml:"enable"`
			Endpoint   string `yaml:"endpoint"`
			AuthToken  string `yaml:"auth_token"`
			ClientCert string `yaml:"client_cert"`
			ClientKey  string `yaml:"client_key"`
			Proxy      string `yaml:"proxy"`
			QueueSize  string `yaml:"queue_size"`
			QueueDir   string `yaml:"queue_dir"`
		} `yaml:"logger_webhook"`
		AuditKafka []struct {
			Name          string `yaml:"name"`
			Enable        string `yaml:"enable"`
			Topic         string `yaml:"topic"`
			Brokers       string `yaml:"brokers"`
			SaslUsername  string `yaml:"sasl_username"`
			SaslPassword  string `yaml:"sasl_password"`
			SaslMechanism string `yaml:"sasl_mechanism"`
			ClientTLSCert string `yaml:"client_tls_cert"`
			ClientTLSKey  string `yaml:"client_tls_key"`
			TLSClientAuth int    `yaml:"tls_client_auth"`
			Sasl          string `yaml:"sasl"`
			TLS           string `yaml:"tls"`
			TLSSkipVerify string `yaml:"tls_skip_verify"`
			Version       string `yaml:"version"`
			QueueSize     int    `yaml:"queue_size"`
			QueueDir      string `yaml:"queue_dir"`
		} `yaml:"audit_kafka"`
		NotifyWebhook []struct {
			Name       string `yaml:"name"`
			Enable     string `yaml:"enable"`
			Endpoint   string `yaml:"endpoint"`
			AuthToken  string `yaml:"auth_token"`
			QueueLimit int    `yaml:"queue_limit"`
			QueueDir   string `yaml:"queue_dir"`
			ClientCert string `yaml:"client_cert"`
			ClientKey  string `yaml:"client_key"`
		} `yaml:"notify_webhook"`
		NotifyAmqp []struct {
			Name              string `yaml:"name"`
			Enable            string `yaml:"enable"`
			URL               string `yaml:"url"`
			Exchange          string `yaml:"exchange"`
			ExchangeType      string `yaml:"exchange_type"`
			RoutingKey        string `yaml:"routing_key"`
			Mandatory         string `yaml:"mandatory"`
			Durable           string `yaml:"durable"`
			NoWait            string `yaml:"no_wait"`
			Internal          string `yaml:"internal"`
			AutoDeleted       string `yaml:"auto_deleted"`
			DeliveryMode      string `yaml:"delivery_mode"`
			PublisherConfirms string `yaml:"publisher_confirms"`
			QueueLimit        int    `yaml:"queue_limit"`
			QueueDir          string `yaml:"queue_dir"`
		} `yaml:"notify_amqp"`
		NotifyKafka []struct {
			Name          string `yaml:"name"`
			Enable        string `yaml:"enable"`
			Topic         string `yaml:"topic"`
			Brokers       string `yaml:"brokers"`
			SaslUsername  string `yaml:"sasl_username"`
			SaslPassword  string `yaml:"sasl_password"`
			SaslMechanism string `yaml:"sasl_mechanism"`
			ClientTLSCert string `yaml:"client_tls_cert"`
			ClientTLSKey  string `yaml:"client_tls_key"`
			TLSClientAuth int    `yaml:"tls_client_auth"`
			Sasl          string `yaml:"sasl"`
			TLS           string `yaml:"tls"`
			TLSSkipVerify string `yaml:"tls_skip_verify"`
			QueueLimit    int    `yaml:"queue_limit"`
			QueueDir      string `yaml:"queue_dir"`
			Version       string `yaml:"version"`
			BatchSize     int    `yaml:"batch_size"`
		} `yaml:"notify_kafka"`
		NotifyMqtt []struct {
			Name              string `yaml:"name"`
			Enable            string `yaml:"enable"`
			Broker            string `yaml:"broker"`
			Topic             string `yaml:"topic"`
			Password          string `yaml:"password"`
			Username          string `yaml:"username"`
			Qos               int    `yaml:"qos"`
			KeepAliveInterval string `yaml:"keep_alive_interval"`
			ReconnectInterval string `yaml:"reconnect_interval"`
			QueueDir          string `yaml:"queue_dir"`
			QueueLimit        int    `yaml:"queue_limit"`
		} `yaml:"notify_mqtt"`
		NotifyNats []struct {
			Name                        string `yaml:"name"`
			Enable                      string `yaml:"enable"`
			Address                     string `yaml:"address"`
			Subject                     string `yaml:"subject"`
			Username                    string `yaml:"username"`
			Password                    string `yaml:"password"`
			Token                       string `yaml:"token"`
			TLS                         string `yaml:"tls"`
			TLSSkipVerify               string `yaml:"tls_skip_verify"`
			CertAuthority               string `yaml:"cert_authority"`
			ClientCert                  string `yaml:"client_cert"`
			ClientKey                   string `yaml:"client_key"`
			PingInterval                int    `yaml:"ping_interval"`
			Jetstream                   string `yaml:"jetstream"`
			Streaming                   string `yaml:"streaming"`
			StreamingAsync              string `yaml:"streaming_async"`
			StreamingMaxPubAcksInFlight int    `yaml:"streaming_max_pub_acks_in_flight"`
			StreamingClusterID          string `yaml:"streaming_cluster_id"`
			QueueDir                    string `yaml:"queue_dir"`
			QueueLimit                  int    `yaml:"queue_limit"`
		} `yaml:"notify_nats"`
		NotifyNsq []struct {
			Name          string `yaml:"name"`
			Enable        string `yaml:"enable"`
			NsqdAddress   string `yaml:"nsqd_address"`
			Topic         string `yaml:"topic"`
			TLS           string `yaml:"tls"`
			TLSSkipVerify string `yaml:"tls_skip_verify"`
			QueueDir      string `yaml:"queue_dir"`
			QueueLimit    int    `yaml:"queue_limit"`
		} `yaml:"notify_nsq"`
		NotifyMysql []struct {
			Name               string `yaml:"name"`
			Enable             string `yaml:"enable"`
			Format             string `yaml:"format"`
			DsnString          string `yaml:"dsn_string"`
			Table              string `yaml:"table"`
			QueueDir           string `yaml:"queue_dir"`
			QueueLimit         int    `yaml:"queue_limit"`
			MaxOpenConnections int    `yaml:"max_open_connections"`
		} `yaml:"notify_mysql"`
		NotifyPostgres []struct {
			Name               string `yaml:"name"`
			Enable             string `yaml:"enable"`
			Format             string `yaml:"format"`
			ConnectionString   string `yaml:"connection_string"`
			Table              string `yaml:"table"`
			QueueDir           string `yaml:"queue_dir"`
			QueueLimit         int    `yaml:"queue_limit"`
			MaxOpenConnections int    `yaml:"max_open_connections"`
		} `yaml:"notify_postgres"`
		NotifyElasticsearch []struct {
			Name       string `yaml:"name"`
			Enable     string `yaml:"enable"`
			URL        string `yaml:"url"`
			Format     string `yaml:"format"`
			Index      string `yaml:"index"`
			QueueDir   string `yaml:"queue_dir"`
			QueueLimit int    `yaml:"queue_limit"`
			Username   string `yaml:"username"`
			Password   string `yaml:"password"`
		} `yaml:"notify_elasticsearch"`
		NotifyRedis []struct {
			Name       string `yaml:"name"`
			Enable     string `yaml:"enable"`
			Format     string `yaml:"format"`
			Address    string `yaml:"address"`
			Key        string `yaml:"key"`
			Password   string `yaml:"password"`
			QueueDir   string `yaml:"queue_dir"`
			QueueLimit int    `yaml:"queue_limit"`
		} `yaml:"notify_redis"`
		LambdaWebhook []struct {
			Name       string `yaml:"name"`
			Enable     string `yaml:"enable"`
			Endpoint   string `yaml:"endpoint"`
			AuthToken  string `yaml:"auth_token"`
			ClientCert string `yaml:"client_cert"`
			ClientKey  string `yaml:"client_key"`
		} `yaml:"lambda_webhook"`
	} `yaml:"options"`
}
```

This is the final data structure that is expected in future.